server/src/main/java/org/opensearch/common/util/TraceUtil.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.util;
+
+/**
+ * Util class to extract trace id from traceparent header
+ */
+public class TraceUtil {
+
+    /**
+     * Extract trace id from traceparent header
+     * @param traceparent traceparent header value
+     * @return trace id
+     */
+    public static String extractTraceId(String traceparent) {
+        if (traceparent == null || traceparent.length() < 55) {
```

Comment:
What if we implement stricter verification logic? For example, we could break down the traceparent into its four distinct segments and perform granular checks on each. these are all very lightweight operations.